restapi/src/main/java/io/stargate/web/docsapi/service/DocsShredder.java

```diff
@@ -208,6 +144,133 @@ private Object convertToBackendBooleanValue(boolean value, boolean numericBoolea
     }
   }
 
+  private ImmutablePair<Map<String, Object>, List<String>> convertJsonNodeToBoundVariables(
+      JsonNode jsonValue,
+      ParsingContext parsingContext,
+      DocumentDB db,
+      List<String> path,
+      String key,
+      boolean patching) {
+    List<String> firstLevelKeys = new ArrayList<>();
+    JsonPath jsonPath =
+        JsonPathCompiler.compile(
+            DocsApiUtils.convertJsonToBracketedPath(parsingContext.getJsonPath()));
+    Map<String, Object> bindMap = db.newBindMap(path);
+
+    bindMap.put("key", key);
+
+    Iterator<PathOperator> it = jsonPath.iterator();
+    String leaf = null;
+    int i = path.size();
+
+    String[] unboundPaths = new String[DocumentDB.MAX_DEPTH];
```

Comment:
can we read the max depth from the configuration? you have it just few lines below